From 72adccf495f5d2b24bcf37c1e1d76a2a182b3e75 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:26 -0400
Subject: [ethernet] qlcnic: include irq.h for irq definitions

Message-id: <1408626422-3941-103-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88662
O-Subject: [PATCH RHEL 7.1 102/138] net: qlcnic: include irq.h for irq definitions
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Josh Boyer <jwboyer@fedoraproject.org>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id acdd32be6d3ec5f872d9fd930cac54d18d6c74ac

The qlcnic driver fails to build on ARM with errors like:

In file included from drivers/net/ethernet/qlogic/qlcnic/qlcnic.h:36:0,
                 from drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c:8:
drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h:585:1: error: unknown type name 'irqreturn_t'
 irqreturn_t qlcnic_83xx_clear_legacy_intr(struct qlcnic_adapter *);
 ^

Nothing in the driver is explicitly including the irq definitions, so we
add an include of linux/irq.h to pick them up.

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
index 24685a0..73f3a3f 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
@@ -23,6 +23,7 @@
 #include <linux/ethtool.h>
 #include <linux/mii.h>
 #include <linux/timer.h>
+#include <linux/irq.h>
 
 #include <linux/vmalloc.h>
 
-- 
1.7.1